tutorials/tech-terms/index.md

```diff
@@ -0,0 +1,283 @@
+
+RPC can be understood as a client-server interaction in which the "caller" is the client, more specifically the requesting program, and the "executor" is the server, more specifically the service-providing program. The interaction is implemented through a request-response message-passing system.
+
+In short, RPC is a request-response protocol, initiated by a client sending a request to a remote server to execute the subroutine. 
+
+RPC allows calling functions in different address spaces. Usually, the called functions are run on a different computer than the one calling them. However, with RPC, the developer codes as if the subroutine would be local; the developer does not have to code in details for remote interaction. Thus, with RPCs it is implied that all calling procedures are basically the same, independent of them being local or remote calls.
+
+<HighlightBox type="note">
+
+As RPCs implement remote request-response protocols, it is important to note that remote procedure calls can fail in case of network problems.
+
+</HighlightBox>
+
+### How does an RPC request work?
+
+In general, when a remote procedure call is invoked the procedure parameters are transferred across the network to the execution environment where the procedure is executed. Once finished, the results of the procedure call invoked are transferred to the call environment. The execution then resumes in the call environment, just as it would in a regular local procedure call.
+
+A step-by-step RPC request could look like the following:
+
+1. A client calls a client stub - a piece of code converting parameters that are passed between client and servers during an RPC. The call is a local procedure call.
+
+<HighlightBox type="info">
+
+A stub is a small program routine substituting a longer program. This allows machines to behave as if a program on a remote machine was operating locally. The client has a stub which interfaces with the remote procedure, while the server has a stub to interface with the original request procedure.
+
+<br></br>
+In RPCs, the client's stub substitutes for the program providing a request procedure. The stub accepts and forwards the request to the remote procedure. Once the remote procedure completes the request, it returns the results to the stub which in turn passes them to the request procedure.
+
+<br></br>
+The server also has a stub to interface with the remote procedure.
+
+</HighlightBox>
+
+2. The client stub packs the procedure parameters into a message.
+
+<HighlightBox type="info">
+
+Packing procedure parameters is called **marshaling**.
+
+<br></br>
+Specifically, this is _the process of gathering data from one or more applications, putting data pieces into a message buffer, and organizing the data into a prescribed data format_.
+
+<br></br>
```

Comment:
When I `npm run serve`, the `<br/>` whitespace is huge. @coldice

Reply:
@CitMC these are two br's - the br tag is a single tag, it cannot be opened and closed. If you want to be strict, you can put `<br/>` instead of `<br>`, but should never use `</br>`.

Reply:
Good to know. I just committed a change that should fix it. @coldice please check before merging.